lib/chcol/variant.go

```diff
@@ -0,0 +1,126 @@
+// Licensed to ClickHouse, Inc. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. ClickHouse, Inc. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package chcol
+
+import (
+	"database/sql/driver"
+	"encoding/json"
+)
+
+// Variant represents a ClickHouse Variant type that can hold multiple possible types
+type Variant struct {
+	value any
+}
+
+// NewVariant creates a new Variant with the given value
+func NewVariant(v any) Variant {
+	return Variant{value: v}
+}
+
+// Nil returns true if the underlying value is nil.
+func (v Variant) Nil() bool {
+	return v.value == nil
+}
+
+// Any returns the underlying value as any. Same as Interface.
+func (v Variant) Any() any {
+	return v.value
+}
+
+// Interface returns the underlying value as interface{}. Same as Any.
+func (v Variant) Interface() interface{} {
+	return v.value
+}
```

Comment:
What is the point? `interface{}` is an alias to `any`.

Reply:
I put both so users could choose whichever name they prefer within their application. Some apps prefer using `any` and others `interface{}`

Reply:
It's only a name. Why have both functions?